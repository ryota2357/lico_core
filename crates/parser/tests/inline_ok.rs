//! This file is generated by `cargo codegen parser`, do not edit by hand.
#![cfg_attr(rustfmt, rustfmt::skip)]

mod utils;
const GROUP: &str = "inline/ok";

#[test]
fn expr_literals() {
    let source = include_str!("../test_data/inline/ok/expr_literals.lico");
    let snapshot = utils::make_snapshot("expr_literals", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn local() {
    let source = include_str!("../test_data/inline/ok/local.lico");
    let snapshot = utils::make_snapshot("local", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn param_list() {
    let source = include_str!("../test_data/inline/ok/param_list.lico");
    let snapshot = utils::make_snapshot("param_list", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn attr_stmt() {
    let source = include_str!("../test_data/inline/ok/attr_stmt.lico");
    let snapshot = utils::make_snapshot("attr_stmt", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn break_stmt() {
    let source = include_str!("../test_data/inline/ok/break_stmt.lico");
    let snapshot = utils::make_snapshot("break_stmt", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn continue_stmt() {
    let source = include_str!("../test_data/inline/ok/continue_stmt.lico");
    let snapshot = utils::make_snapshot("continue_stmt", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn for_stmt() {
    let source = include_str!("../test_data/inline/ok/for_stmt.lico");
    let snapshot = utils::make_snapshot("for_stmt", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn func() {
    let source = include_str!("../test_data/inline/ok/func.lico");
    let snapshot = utils::make_snapshot("func", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn return_stmt() {
    let source = include_str!("../test_data/inline/ok/return_stmt.lico");
    let snapshot = utils::make_snapshot("return_stmt", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn var_stmt() {
    let source = include_str!("../test_data/inline/ok/var_stmt.lico");
    let snapshot = utils::make_snapshot("var_stmt", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn while_stmt() {
    let source = include_str!("../test_data/inline/ok/while_stmt.lico");
    let snapshot = utils::make_snapshot("while_stmt", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
