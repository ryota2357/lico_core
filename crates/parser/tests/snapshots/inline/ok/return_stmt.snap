---
source: crates/parser/tests/inline_ok.rs
---
# return_stmt

## Input

```
return
return 1
```

## CST

```
SOURCE_FILE@0..15
  RETURN_STMT@0..6
    RETURN_KW@0..6 "return"
  WHITESPACE@6..7 "\n"
  RETURN_STMT@7..15
    RETURN_KW@7..13 "return"
    WHITESPACE@13..14 " "
    LITERAL@14..15
      INT@14..15 "1"
```

## AST

```
SourceFile {
    shebang: none,
    statements: [
        ReturnStmt {
            return_token: RETURN_KW@0..6 "return",
            expr: none,
        },
        ReturnStmt {
            return_token: RETURN_KW@7..13 "return",
            expr: Literal {
                token: INT@14..15 "1",
            },
        },
    ],
}
```

vim:ft=markdown
